packages/api/src/utils/client/client.ts

```diff
@@ -26,7 +26,7 @@ export function getFetchOptsSerializer<Fn extends (...args: any) => any, ReqType
     return {
       url: urlFormater(req.params ?? {}),
       method: routeDef.method,
-      query: req.query,
+      query: Object.keys(req.query ?? {}).length ? req.query : undefined,
```

Comment:
Fixes appending '?' to URL if query params are omitted, e.g. `/eth/v1/node/health` instead of `/eth/v1/node/health?`.

I would assume most/(all) HTTP server will ignore this anyways, but it might be misleading if req.url is logged on the server.